### DISCUSSION TITLE
Pitch: Enhancing events emitted during Round creation on v1.0.0

### DISCUSSION THREAD

**Discussion by lmcorbalan** [allo-protocol/aips#38] [New Ideas]
#### Problem
- **Registry.sol**: When a round is created it emits 3 types of events in the following order: `RoleGranted` -> `MetadataUpdated` -> `ProjectCreated` . This order is not useful when working on the subgraph and indexer.
- **RoundFactory.sol**: The `VotingContractCreated` event contains the address of the newly created round and strategy contracts (both are clones) but no strategy implementation address. 

#### Suggested Implementation
- Reorder event emission on `Registry.sol` to `ProjectCreated` -> `RoleGranted` -> `MetadataUpdated`.
- Add the address of the implementation used for creating the strategy contract to the `VotingContractCreated` event.
- I would also suggest changing the event name `VotingContractCreated` to `StrategyContractCreated`

Reference implementation https://github.com/BootNodeDev/allo-protocol-contracts/pull/5

**Comment by davejkwm** [comment]
Looks good
